test/e2e/status_manager.go

```diff
@@ -0,0 +1,120 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2ekubectl "k8s.io/kubernetes/test/e2e/framework/kubectl"
+)
+
+var _ = ginkgo.Describe("Status manager validation", func() {
+	const (
+		svcname                string = "status-manager"
+		egressFirewallYamlFile string = "egress-fw.yml"
+	)
+
+	f := wrappedTestFramework(svcname)
+
+	ginkgo.BeforeEach(func() {
+		// create EgressFirewall
+		var egressFirewallConfig = fmt.Sprintf(`kind: EgressFirewall
+apiVersion: k8s.ovn.org/v1
+metadata:
+  name: default
+  namespace: %s
+spec:
+  egress:
+  - type: Allow
+    to:
+      cidrSelector: 1.2.3.4/24
+`, f.Namespace.Name)
+		// write the config to a file for application and defer the removal
+		if err := os.WriteFile(egressFirewallYamlFile, []byte(egressFirewallConfig), 0644); err != nil {
+			framework.Failf("Unable to write CRD config to disk: %v", err)
+		}
+		// create the CRD config parameters
+		applyArgs := []string{
+			"apply",
+			fmt.Sprintf("--namespace=%s", f.Namespace.Name),
+			"-f",
+			egressFirewallYamlFile,
+		}
+		framework.Logf("Applying EgressFirewall configuration: %s ", applyArgs)
+		// apply the egress firewall configuration
+		e2ekubectl.RunKubectlOrDie(f.Namespace.Name, applyArgs...)
+		// check status
+		checkEgressFirewallStatus(f.Namespace.Name, false, true, true)
+	})
+
+	ginkgo.AfterEach(func() {
+		if err := os.Remove(egressFirewallYamlFile); err != nil {
+			framework.Logf("Unable to remove the CRD config from disk: %v", err)
+		}
+	})
+
+	ginkgo.It("Should validate the egress firewall status when adding an unknown zone", func() {
```

Comment:
is this a way to test status manager through egress firewall status, or are we missing tests for APB route?

Reply:
yeah this one is intended for status manager testing, doesn't really matter which object to use.
I will try to add all resource-specific tests to unit tests.

Reply:
now I think I probably will add 1 extra test for every resource to catch potential rbac problems downstream

Reply:
morning thoughts: we don't need to add resource-specific tests, because both egress firewall and apbroute e2e tests check their statuses to figure out when the config is applied (and hopefully the same will apply to the other resources in the future, at least it seems to be the only way to avoid races)